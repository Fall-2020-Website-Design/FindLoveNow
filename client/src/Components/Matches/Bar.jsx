import React, { Component } from 'react';
import Row from 'react-bootstrap/Row';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.css';
import './Matches.css'

const SingleBar = styled.button`
    height: 7px;
    width: ${props => props.size};
    margin-left: 5px;
    background: ${props => props.color};
`;

export class Bar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            total: this.props.total,
            bars: [],
            barSize: (30 / this.props.total).toString() + "rem"
        }
    }

    componentDidMount () {
        let bar = []
        
        console.log(this.state.barSize)
        console.log(window.innerWidth )
        
        for(let i=0; i < this.state.total; i++) {
            bar.push(
                <SingleBar key={i} size={this.state.barSize} color={0 === i ? "#FF0000" : "#808080"} onClick={() => this.props.click(i)}/>
            );
        }

        this.setState({
            bars: bar
        })
    }

    componentWillReceiveProps (newProps) {
        let bar = []

        for(let i=0; i < this.state.total; i++) {
            bar.push(
                <SingleBar key={i} size={this.state.barSize} color={newProps.current === i ? "#FF0000" : "#808080"} onClick={() => this.props.click(i)}/>
            );
        }

        this.setState({
            bars: bar
        })
    }

    render() {
        return (
            <div>
                <Row className="justify-content-center mt-1" >
                    {this.state.bars}
                </Row>
            </div>
        )
    }
}

export default Bar
