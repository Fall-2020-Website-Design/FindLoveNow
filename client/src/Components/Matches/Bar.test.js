import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Bar } from './Bar';


Enzyme.configure({ adapter: new Adapter() });

describe('Bar', () => {

	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Bar  />);
    });
    
    test('renders', () => {
		const component = renderer.create(
			<MemoryRouter>
				<Bar  />
			</MemoryRouter>
		).toJSON();
		
		expect(component).toMatchSnapshot();
    });
    

});