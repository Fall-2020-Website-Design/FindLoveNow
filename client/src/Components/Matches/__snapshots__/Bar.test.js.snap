// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders 1`] = `
<div>
  <div
    className="justify-content-center mt-1 row"
  />
</div>
`;
